lib/rosette/integrations/smartling-integration/smartling_completer.rb

```diff
@@ -1,49 +1,164 @@
 # encoding: UTF-8
 
+require 'thread'
+require 'concurrent'
+
 module Rosette
   module Integrations
     class SmartlingIntegration < Integration
       class SmartlingCompleter
 
-        attr_reader :configuration, :smartling_apis, :pullers
+        DEFAULT_THREAD_POOL_SIZE = 10
+
+        attr_reader :rosette_config, :repo_config
+        attr_reader :thread_pool_size
```

Comment:
Are you thinking they should be accessors instead? The `set_` methods enable chaining because they return `self`, and the readers allow reading the values later.